app/assets/javascripts/covid19/src/webapp/components/SVGProteoma.tsx

```diff
@@ -0,0 +1,498 @@
+import React from "react";
+import styled from "styled-components";
+import _ from "lodash";
+import i18n from "../../utils/i18n";
+import { sendAnalytics } from "../../utils/analytics";
+
+interface SVGProteomaProps {
+    search: string;
+    setSearch: (value: string) => void;
+    visible: { orf1a?: boolean; orf1b?: boolean };
+    setVisible: (visible: { orf1a?: boolean; orf1b?: boolean }) => void;
+    title: React.ReactNode;
+    setTitle: (value: React.ReactNode) => void;
+    setProteomaSelected: (value: boolean) => void;
+}
+
+export const SVGProteoma: React.FC<SVGProteomaProps> = React.memo(props => {
+    const { search, setSearch, visible, setVisible, title, setTitle, setProteomaSelected } = props;
+    const [loading, setLoading] = React.useState(false);
+    const stateSetters = { setSearch, setTitle, setProteomaSelected, setLoading };
+
+    return (
+        <Container>
+            {/*Using relative -> absolute for having svg above title. So the title can be easily put in center*/}
+            <Layer>{title}</Layer>
+            <Layer>
+                <SVG xmlns="http://www.w3.org/2000/svg" viewBox="0 0 1000 1000">
+                    {/*Wraps all the graphics*/}
+                    <g
+                        onMouseLeave={() => {
+                            setTitle(
+                                <span>
+                                    {i18n.t("SARS-CoV-2")}
+                                    <br />
+                                    {i18n.t("Proteome")}
+                                </span>
+                            );
+                            setVisible({ orf1a: false, orf1b: false });
+                        }}
+                    >
+                        <rect className="none" width="1000" height="1000" />
+                        {/*Wraps ORF1a and its proteins*/}
+                        <g
+                            onMouseEnter={() => {
+                                setVisible({ orf1a: true, orf1b: false });
+                            }}
+                        >
+                            <ProteomaPath
+                                stateSetters={stateSetters}
+                                name="ORF1a"
+                                classStyle="orf1a"
+                                def="M754,500a256.17,256.17,0,0,1-2.84,39.79A243.33,243.33,0,0,1,739,585.89l-45-16.4a204.32,204.32,0,0,0,9.52-37.28A208.29,208.29,0,0,0,706,500c0-2.5,0-4.92-.13-7.21A205.82,205.82,0,0,0,501,294V246a255.53,255.53,0,0,1,47.43,4.61,252.48,252.48,0,0,1,89.88,36.32A253.31,253.31,0,0,1,753.84,491.11C754,493.94,754,496.92,754,500Z"
+                            />
+                            {visible.orf1a && (
```

Comment:
The component is huuuge. Let's split it a little bit. It's usually ok, as we discussed, to add internally used components in the same file, but when a component grows that much, I think it's more clear to create another file. You may create a folder to contain them all: `components/proteome/SVGProteome.tsx` + all inner components.